service/core.go

```diff
@@ -63,20 +65,49 @@ type SetContainerer interface {
 	SetContainer(Host)
 }
 
+type metricsCore struct {
+	scope           tally.RootScope
+	runtimeReporter *metrics.RuntimeReporter
+}
+
+func (mc *metricsCore) Metrics() tally.Scope {
+	return mc.scope
+}
+
+func (mc *metricsCore) RuntimeMetrics() *metrics.RuntimeReporter {
+	return mc.runtimeReporter
+}
+
+type tracerCore struct {
+	tracer       opentracing.Tracer
+	tracerCloser io.Closer
+	tracerConfig jaegerconfig.Configuration
+}
+
+func (tc *tracerCore) Tracer() opentracing.Tracer {
+	return tc.tracer
+}
+
+type loggingCore struct {
```

Comment:
We can go a little further and move logging initialization code here or under /service/core package along with other *Core stuff, and use function call in the service New. service.New is about 100+ lines of code right now.